<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
<<<<<<< ours
diff --git a/pkg/test/framework/components/echo/kube/deployment.go b/pkg/test/framework/components/echo/kube/deployment.go
index 6bc7cfcfd7..f384dcd289 100644
--- a/pkg/test/framework/components/echo/kube/deployment.go
+++ b/pkg/test/framework/components/echo/kube/deployment.go
@@ -520,6 +520,7 @@ spec:
 		if rev := getIstioRevision(cfg.Namespace); len(rev) > 0 {
 			cmd = append(cmd, "--revision", rev)
 		}
+		cmd = append(cmd, "--ingressIP", istiodAddr.Addr().String())
 		// make sure namespace controller has time to create root-cert ConfigMap
 		if err := retry.UntilSuccess(func() error {
 			stdout, stderr, err := istioCtl.Invoke(cmd)
diff --git a/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml b/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
index ea587e4e8f..4572527594 100644
--- a/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
+++ b/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
@@ -87,16 +87,16 @@ spec:
 
           # since we're not overwriting /etc/hosts on k8s, verify that istiod hostname in /etc/hosts
           # matches the value generated by istioctl
-          echo "checking istio host"
-          SYSTEM_HOST=$(cat /etc/hosts | grep istiod)
-          ISTIOCTL_HOST=$(cat /var/run/secrets/istio/bootstrap/hosts | grep istiod)
-          if [ "$(echo "$SYSTEM_HOST" | tr -d '[:space:]')" != "$(echo "$ISTIOCTL_HOST" | tr -d '[:space:]')" ]; then
-            echo "istiod host in /etc/hosts does not match value generated by istioctl"
-            echo "/etc/hosts: $SYSTEM_HOST"
-            echo "/var/run/secrets/istio/bootstrap/hosts: $ISTIOCTL_HOST"
-            exit 1
-          fi
-          echo "istiod host ok"
+          # echo "checking istio host"
+          # SYSTEM_HOST=$(cat /etc/hosts | grep istiod)
+          # ISTIOCTL_HOST=$(cat /var/run/secrets/istio/bootstrap/hosts | grep istiod)
+          # if [ "$(echo "$SYSTEM_HOST" | tr -d '[:space:]')" != "$(echo "$ISTIOCTL_HOST" | tr -d '[:space:]')" ]; then
+          #   echo "istiod host in /etc/hosts does not match value generated by istioctl"
+          #   echo "/etc/hosts: $SYSTEM_HOST"
+          #   echo "/var/run/secrets/istio/bootstrap/hosts: $ISTIOCTL_HOST"
+          #   exit 1
+          # fi
+          # echo "istiod host ok"
 
           # read certs from correct directory
           sudo sh -c 'echo PROV_CERT=/var/run/secrets/istio >> /var/lib/istio/envoy/cluster.env'
diff --git a/pkg/test/framework/components/istio/util.go b/pkg/test/framework/components/istio/util.go
index 2600fe421c..2d01c756c1 100644
--- a/pkg/test/framework/components/istio/util.go
+++ b/pkg/test/framework/components/istio/util.go
@@ -137,6 +137,15 @@ func getRemoteServiceAddress(s *kube.Settings, cluster cluster.Cluster, ns, labe
 	if ingr.IP == "" && ingr.Hostname == "" {
 		return nil, false, fmt.Errorf("service %s/%s is not available yet: no ingress", svc.Namespace, svc.Name)
 	}
+	if ingr.Hostname != "" {
+		ip, err := net.LookupIP(ingr.Hostname)
+		if err != nil {
+			return nil, false, fmt.Errorf("service %s/%s is not available yet: no ingress", svc.Namespace, svc.Name)
+		}
+		if len(ip) > 0 {
+			ingr.IP = ip[0].String()
+		}
+	}
 	if ingr.IP != "" {
 		ipaddr, err := netip.ParseAddr(ingr.IP)
 		if err != nil {
diff --git a/tests/integration/pilot/ingress_test.go b/tests/integration/pilot/ingress_test.go
index 258dcf8de0..2caaef97eb 100644
--- a/tests/integration/pilot/ingress_test.go
+++ b/tests/integration/pilot/ingress_test.go
@@ -230,8 +230,9 @@ spec:
 						retry.UntilSuccessOrFail(t, func() error {
 							gwc, err := t.Clusters().Kube().Default().GatewayAPI().GatewayV1beta1().GatewayClasses().Get(context.Background(), "istio", metav1.GetOptions{})
 							if err != nil {
-								return err
+                                                           return err
 							}
+
 							if s := kstatus.GetCondition(gwc.Status.Conditions, string(k8s.GatewayClassConditionStatusAccepted)).Status; s != metav1.ConditionTrue {
 								return fmt.Errorf("expected status %q, got %q", metav1.ConditionTrue, s)
 							}
@@ -601,6 +602,12 @@ spec:
 						if hostIsIP {
 							got = ing.Status.LoadBalancer.Ingress[0].IP
 						}
+						if ing.Status.LoadBalancer.Ingress[0].Hostname != "" {
+							ip, _ := net.LookupIP(ing.Status.LoadBalancer.Ingress[0].Hostname)
+							if len(ip) > 0 {
+								got = ip[0].String()
+							}
+						}
 						if got != host {
 							return fmt.Errorf("unexpected ingress status, got %+v want %v", got, host)
 						}
@@ -617,6 +624,12 @@ spec:
 					if hostIsIP {
 						got = ing.Status.LoadBalancer.Ingress[0].IP
 					}
+					if ing.Status.LoadBalancer.Ingress[0].Hostname != "" {
+						ip, _ := net.LookupIP(ing.Status.LoadBalancer.Ingress[0].Hostname)
+						if len(ip) > 0 {
+							got = ip[0].String()
+						}
+					}
 					if got != host {
 						return fmt.Errorf("unexpected ingress status, got %+v want %v", got, host)
 					}
diff --git a/tetrateci/patches/eks/eks-ingress.1.16.patch b/tetrateci/patches/eks/eks-ingress.1.16.patch
index bc877cf57f..9f623ae3e3 100644
--- a/tetrateci/patches/eks/eks-ingress.1.16.patch
+++ b/tetrateci/patches/eks/eks-ingress.1.16.patch
@@ -1,103 +0,0 @@
-diff --git a/pkg/test/framework/components/echo/kube/deployment.go b/pkg/test/framework/components/echo/kube/deployment.go
-index 6bc7cfcfd7..f384dcd289 100644
---- a/pkg/test/framework/components/echo/kube/deployment.go
-+++ b/pkg/test/framework/components/echo/kube/deployment.go
-@@ -520,6 +520,7 @@ spec:
- 		if rev := getIstioRevision(cfg.Namespace); len(rev) > 0 {
- 			cmd = append(cmd, "--revision", rev)
- 		}
-+		cmd = append(cmd, "--ingressIP", istiodAddr.Addr().String())
- 		// make sure namespace controller has time to create root-cert ConfigMap
- 		if err := retry.UntilSuccess(func() error {
- 			stdout, stderr, err := istioCtl.Invoke(cmd)
-diff --git a/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml b/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
-index ea587e4e8f..4572527594 100644
---- a/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
-+++ b/pkg/test/framework/components/echo/kube/templates/vm_deployment.yaml
-@@ -87,16 +87,16 @@ spec:
- 
-           # since we're not overwriting /etc/hosts on k8s, verify that istiod hostname in /etc/hosts
-           # matches the value generated by istioctl
--          echo "checking istio host"
--          SYSTEM_HOST=$(cat /etc/hosts | grep istiod)
--          ISTIOCTL_HOST=$(cat /var/run/secrets/istio/bootstrap/hosts | grep istiod)
--          if [ "$(echo "$SYSTEM_HOST" | tr -d '[:space:]')" != "$(echo "$ISTIOCTL_HOST" | tr -d '[:space:]')" ]; then
--            echo "istiod host in /etc/hosts does not match value generated by istioctl"
--            echo "/etc/hosts: $SYSTEM_HOST"
--            echo "/var/run/secrets/istio/bootstrap/hosts: $ISTIOCTL_HOST"
--            exit 1
--          fi
--          echo "istiod host ok"
-+          # echo "checking istio host"
-+          # SYSTEM_HOST=$(cat /etc/hosts | grep istiod)
-+          # ISTIOCTL_HOST=$(cat /var/run/secrets/istio/bootstrap/hosts | grep istiod)
-+          # if [ "$(echo "$SYSTEM_HOST" | tr -d '[:space:]')" != "$(echo "$ISTIOCTL_HOST" | tr -d '[:space:]')" ]; then
-+          #   echo "istiod host in /etc/hosts does not match value generated by istioctl"
-+          #   echo "/etc/hosts: $SYSTEM_HOST"
-+          #   echo "/var/run/secrets/istio/bootstrap/hosts: $ISTIOCTL_HOST"
-+          #   exit 1
-+          # fi
-+          # echo "istiod host ok"
- 
-           # read certs from correct directory
-           sudo sh -c 'echo PROV_CERT=/var/run/secrets/istio >> /var/lib/istio/envoy/cluster.env'
-diff --git a/pkg/test/framework/components/istio/util.go b/pkg/test/framework/components/istio/util.go
-index 2600fe421c..2d01c756c1 100644
---- a/pkg/test/framework/components/istio/util.go
-+++ b/pkg/test/framework/components/istio/util.go
-@@ -137,6 +137,15 @@ func getRemoteServiceAddress(s *kube.Settings, cluster cluster.Cluster, ns, labe
- 	if ingr.IP == "" && ingr.Hostname == "" {
- 		return nil, false, fmt.Errorf("service %s/%s is not available yet: no ingress", svc.Namespace, svc.Name)
- 	}
-+	if ingr.Hostname != "" {
-+		ip, err := net.LookupIP(ingr.Hostname)
-+		if err != nil {
-+			return nil, false, fmt.Errorf("service %s/%s is not available yet: no ingress", svc.Namespace, svc.Name)
-+		}
-+		if len(ip) > 0 {
-+			ingr.IP = ip[0].String()
-+		}
-+	}
- 	if ingr.IP != "" {
- 		ipaddr, err := netip.ParseAddr(ingr.IP)
- 		if err != nil {
-diff --git a/tests/integration/pilot/ingress_test.go b/tests/integration/pilot/ingress_test.go
-index 258dcf8de0..96e7cc04cd 100644
---- a/tests/integration/pilot/ingress_test.go
-+++ b/tests/integration/pilot/ingress_test.go
-@@ -228,8 +228,7 @@ spec:
- 					})
- 					t.NewSubTest("status").Run(func(t framework.TestContext) {
- 						retry.UntilSuccessOrFail(t, func() error {
--							gwc, err := t.Clusters().Kube().Default().GatewayAPI().GatewayV1beta1().GatewayClasses().Get(context.Background(), "istio", metav1.GetOptions{})
--							if err != nil {
-+							gwc, err := t.Clusters().Kube().Default().GatewayAPI().GatewayV1alpha2().GatewayClasses().Get(context.Background(), "istio", metav1.GetOptions{})							if err != nil {
- 								return err
- 							}
- 							if s := kstatus.GetCondition(gwc.Status.Conditions, string(k8s.GatewayClassConditionStatusAccepted)).Status; s != metav1.ConditionTrue {
-@@ -601,6 +600,12 @@ spec:
- 						if hostIsIP {
- 							got = ing.Status.LoadBalancer.Ingress[0].IP
- 						}
-+						if ing.Status.LoadBalancer.Ingress[0].Hostname != "" {
-+							ip, _ := net.LookupIP(ing.Status.LoadBalancer.Ingress[0].Hostname)
-+							if len(ip) > 0 {
-+								got = ip[0].String()
-+							}
-+						}
- 						if got != host {
- 							return fmt.Errorf("unexpected ingress status, got %+v want %v", got, host)
- 						}
-@@ -617,6 +622,12 @@ spec:
- 					if hostIsIP {
- 						got = ing.Status.LoadBalancer.Ingress[0].IP
- 					}
-+					if ing.Status.LoadBalancer.Ingress[0].Hostname != "" {
-+						ip, _ := net.LookupIP(ing.Status.LoadBalancer.Ingress[0].Hostname)
-+						if len(ip) > 0 {
-+							got = ip[0].String()
-+						}
-+					}
- 					if got != host {
- 						return fmt.Errorf("unexpected ingress status, got %+v want %v", got, host)
- 					}
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
=======
>>>>>>> theirs
